codes/quantum/oscillators/fock_state/knill-laflamme-milburn-loqc.yml

```diff
@@ -0,0 +1,71 @@
+physical: qubits
+logical: qubits
+
+name: 'Knill-Laflamme-Milburn Linear Optical QC encoding'
+introduced: '\cite{doi:10.1038/35051009}' 
+
+
+description: 'Qubits are encoded with dual-rail logic i.e. using the physical location of a photon in two spatial modes. Logical \(|\bar{0}\rangle\) is encoded as \(|01\rangle_{ab}\) and logical \(|\bar{1}\rangle\) is encoded as \(|10\rangle_{ab}\), where a and b are spatial modes. By performing non-linear gates adaptively, the Knill-Laflamme-Milburn protocol approximates linear gates for universal quantum computation. For scalability, the protocol is enhanced by using teleportation to increase success probability of the adaptive circuit.
+
+The KLM protocol is further enhanced by applying explicit error-correction after teleportation, by concatenating an outer stabilizer code with the dual-rail logic code. Logical \(|\bar{0}\rangle\) is encoded as \(\left|\Phi^{+}\right\rangle\), and logical \(|\bar{1}\rangle\) is encoded as \(\left|\Psi^{+}\right\rangle\).'
+
+
+protection: 'Protects against an effective projective measurement in the Z-basis that occurs when teleportation fails i.e. when the Bell measurement yields 0 or 2 photons. During the teleportation of a 2-qubit gate, the intermediate Bell-measurement step takes place through photo-detection, which requires detecting exactly 1 photon. When we measure 0 or 2 photons, we obtain \(|\bar{0}\rangle\) and \(|\bar{1}\rangle\) respectively and hence effectively measure the logical qubit in the Z-basis. \cite{arXiv:quant-ph/0512104}'
+
+
+features:
+
+  encoders:
+    - 'Qubits are first encoded with dual-rail logic i.e. using the physical location of a photon in two spatial modes. Logical \(|\bar{0}\rangle\) is encoded as \(|01\rangle_{ab}\) and logical \(|\bar{1}\rangle\) is encoded as \(|10\rangle_{ab}\), where a and b are spatial modes.'
+    - 'The dual-rail logic encoding is isomorphic to encoding in the horizontal and vertically polarized basis, up to a transformation using a polarising beam-splitter and polarising rotator.'
+    - 'The dual-rail code is concatenated with an outer stabilizer code, with the following two codewords: Logical \(|\bar{0}\rangle\) is encoded as \(\left|\Phi^{+}\right\rangle = \frac{1}{2}(|00\rangle_q + |11\rangle_q)\), and logical \(|\bar{1}\rangle\) is encoded as \(\left|\Psi^{+}\right\rangle  = \frac{1}{2}(|01\rangle_q + |10\rangle_q)\).'
+
+  transversal_gates: 'The KLM protocol relies on single-qubit unitaries or controlled-Z gates, which are transversal. Hence the encoded operations can be performed transversally.'
+
+  general_gates:
+    - 'Single-qubit unitaries can be performed using purely linear optical elements. This is because any single qubit unitary U can be decomposed into rotations about the Z and Y axis in the Bloch sphere i.e. \( U=e^{i \alpha} R_{z}(\beta) R_{y}(\gamma) R_{z}(\delta)\). Arbitary Z and Y rotations which can be achieved using phase-shifters and beam splitters.'
```

Comment:
I'm not sure we need the detail about the decomposition of single-qubit unitaries into Z and Y rotations.